src/home/homework.c

```diff
@@ -519,9 +519,13 @@ static int read_identity_file(int root_fd, JsonVariant **ret) {
         assert(root_fd >= 0);
         assert(ret);
 
-        identity_fd = openat(root_fd, ".identity", O_RDONLY|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW|O_NONBLOCK);
+        identity_fd = openat(root_fd, ".identity/record.json", O_RDONLY|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW|O_NONBLOCK);
+        if (identity_fd < 0 && errno == ENOTDIR) {
```

Comment:
see, this is the problem here: we will follow the symlink on `.identity/` here, which can lead anywhere. For example people can do `ln -s /home/adrian/.identity/record.json /home/lennart/.identity/record.json` and we'd just eat this up as if it was nothging. 

That's a major security hole. In the previous logic this didn't exist, since we'd read a file in the top-level dir of the homedir only, and do so with O_NOFOLLOW, so we'd never follow a single symlink anywhere else, we cannot be tricked into reading data we shouldn't read.

Now, we have in the chase() apis a logic to deal with symlink issues like this, but anyone else doing this for example from shell would sitll be vulnerable. I don#t think we should build security pitfalls like this into our code.

Reply:
Well no `ln -s /home/adrian/.identity/record.json /home/lennart/.identity/record.json` would fail. But someone can symlink their entire ~/.identity folder somewhere else and cause issues that way yes.

Adding yet another directory into the top-level home directory really isn't something I want to do. I get annoyed whenever random software does so, and I don't really see a reason why homed should be an exception to this. I think systemd just needs to be careful not to follow a symlink here, and we should document that other software should be careful too, and call it at that.

Honestly, IMO _nothing else_ should touch homed's record. If anything we should make .identity owned by 0:0. This is data owned by homed and really doesn't belong to the user. We only store it in the home directory so that homed can reconcile things if the home directory is moved between systems. If someone else touches it IMO they're on their own, and if they're doing so in a language like bash then they're already covered in footguns.

If in 5 years we have something else that homed wants to store in the home directory, adding a third/fourth/fitfth/nth file directly in the directory seems misguided.